packages/webpack-plugin/src/runtime-inject.ts

```diff
@@ -0,0 +1,195 @@
+import { stylesheet, injectStyles } from './runtime';
+
+import {
+    RuntimeModule,
+    RuntimeGlobals,
+    Compilation,
+    Dependency,
+    Module,
+    ModuleGraph,
+    NormalModule,
+    sources,
+    ChunkGraph,
+} from 'webpack';
+import { getStylableBuildMeta, replaceMappedCSSAssetPlaceholders } from './plugin-utils';
+import {
+    DependencyTemplates,
+    RuntimeTemplate,
+    StringSortableSet,
+    StylableBuildMeta,
+} from './types';
+const makeSerializable = require('webpack/lib/util/makeSerializable');
+
+interface DependencyTemplateContext {
+    module: Module;
+    moduleGraph: ModuleGraph;
+    runtimeRequirements: Set<string>;
+    runtimeTemplate: RuntimeTemplate;
+    runtime?: string | StringSortableSet;
+    chunkGraph: ChunkGraph;
+    dependencyTemplates: DependencyTemplates;
+}
+
+export class StylableRuntimeDependency extends Dependency {
+    constructor(private stylableBuildMeta: StylableBuildMeta) {
+        super();
+    }
+    updateHash(hash: any) {
+        hash.update(JSON.stringify(this.stylableBuildMeta));
+    }
+
+    serialize({ write }: any) {
+        write(this.stylableBuildMeta);
+    }
+
+    deserialize({ read }: any) {
+        this.stylableBuildMeta = read();
+    }
+}
+
+export class InjectDependencyTemplate {
+    constructor(
+        private staticPublicPath: string,
+        private assetsModules: Map<string, NormalModule>,
+        private runtimeStylesheetId: 'namespace' | 'module',
+        private runtimeId: string
+    ) {}
+    apply(
+        _dependency: StylableRuntimeDependency,
+        source: sources.ReplaceSource,
+        {
+            module,
+            runtimeRequirements,
+            runtimeTemplate,
+            moduleGraph,
+            runtime,
+            chunkGraph,
+            dependencyTemplates,
+        }: DependencyTemplateContext
+    ) {
+        const stylableBuildMeta = getStylableBuildMeta(module);
+        if (!stylableBuildMeta.isUsed) {
+            return;
+        }
+        if (stylableBuildMeta.cssInjection === 'js') {
+            const css = replaceMappedCSSAssetPlaceholders({
+                assetsModules: this.assetsModules,
+                staticPublicPath: this.staticPublicPath,
+                chunkGraph,
+                moduleGraph,
+                dependencyTemplates,
+                runtime,
+                runtimeTemplate,
+                stylableBuildMeta,
+            });
+
+            if (!(module instanceof NormalModule)) {
```

Comment:
this can easily break if stylable's webpack-plugin gets a different copy of webpack. if there aren't many places like it, we may want to replace it into a looser check (constructor.name fallback, etc).

Reply:
true. I'm not sure if anything should work if there is such a case because we extends webpack internals and webpack internally checks for `instanceof`